pylint/checkers/strings.py

```diff
@@ -541,15 +549,45 @@ class StringConstantChecker(BaseTokenChecker):
     # Unicode strings.
     UNICODE_ESCAPE_CHARACTERS = 'uUN'
 
+    def __init__(self, *args, **kwargs):
+        super(StringConstantChecker, self).__init__(*args, **kwargs)
+        self.string_tokens = {}
+
     def process_module(self, module):
         self._unicode_literals = 'unicode_literals' in module.future_imports
 
     def process_tokens(self, tokens):
-        for (tok_type, token, (start_row, _), _, _) in tokens:
+        for (tok_type, token, start, _, _) in tokens:
             if tok_type == tokenize.STRING:
                 # 'token' is the whole un-parsed token; we can look at the start
                 # of it to see whether it's a raw or unicode string etc.
-                self.process_string_token(token, start_row)
+                self.process_string_token(token, start[0])
+                try:
+                    self.string_tokens[start] = ast.literal_eval(token)
```

Comment:
What's the purpose of literal_evaling these tokens?

Reply:
To get a string out of a token string.
There are some examples of token values:
- `"'a'"``
- `"u'b'"`
- `"r'b'"`
- `"'b \\\n              c'"`